Libs/MRML/Core/vtkMRMLSubjectHierarchyNode.cxx

```diff
@@ -2599,6 +2599,21 @@ void vtkMRMLSubjectHierarchyNode::SetItemParent(vtkIdType itemID, vtkIdType pare
     return;
     }
 
+  // Check if the new parent is not a child of the item
+  if (!skipCircularCheck)
+    {
+    vtkSubjectHierarchyItem* currentParentItem = parentItem;
+    while ( (currentParentItem = currentParentItem->Parent) != nullptr )
```

Comment:
If there is already a loop then this check will hang. As it is a safety check, it would be probably nice to report if there is a loop already. We can detect an existing loop by comparing "item" to "parentItem" as well.

Reply:
I think the new version should do it. Let me know.